src/Umbraco.Cms.Imaging.ImageSharp/ConfigureImageSharpMiddlewareOptions.cs

```diff
@@ -34,29 +34,27 @@ public void Configure(ImageSharpMiddlewareOptions options)
     {
         options.Configuration = _configuration;
 
+        options.HMACSecretKey = _imagingSettings.HMACSecretKey;
         options.BrowserMaxAge = _imagingSettings.Cache.BrowserMaxAge;
         options.CacheMaxAge = _imagingSettings.Cache.CacheMaxAge;
         options.CacheHashLength = _imagingSettings.Cache.CacheHashLength;
 
         // Use configurable maximum width and height
         options.OnParseCommandsAsync = context =>
         {
-            if (context.Commands.Count == 0)
+            if (context.Commands.Count == 0 || _imagingSettings.HMACSecretKey.Length > 0)
```

Comment:
We can safely skip validating the maximum width/height when using HMAC authentication (the default callback in V2 also removed the hardcoded width/height check).